lib/streamlit/app_session.py

```diff
@@ -377,12 +382,9 @@ def _enqueue_file_change_message(self) -> None:
         self.enqueue(msg)
 
     def _enqueue_new_session_message(self) -> None:
-        new_id = generate_new_id()
-        self._session_data.script_run_id = new_id
-
         msg = ForwardMsg()
 
-        msg.new_session.script_run_id = self._session_data.script_run_id
+        msg.new_session.script_run_id = _generate_scriptrun_id()
```

Comment:
We could have `_enqueue_new_session_message` take an optional script run id to make testing a little nicer, but since we only have one test that uses it at all it doesn't matter much.